STP-3353: Add info on changing BOS version to integration branch

jld-dehn-hpe opened the pull request:
## Summary and Scope

Added information on changing the BOS version. You can select the BOS version to use for individual commands with the `--bos-version` option. You can also configure the BOS version to use in the SAT config file.

## Issues and Related PRs

* Resolves [STP-3353](https://jira-pro.its.hpecorp.net:8443/browse/STP-3353)
* Change will also be needed in `release/2.4`

## Testing

Spell check and lint check

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

ekamin-hpe reviewed (approved):
Looks good to me. I just have one small comment.

It looks like there is an issue with the build right now but it looks like a temporary hiccup, will probably go away when you push new changes.

haasken-hpe commented on docs/usage/change_bos_version.md:
````diff
@@ -0,0 +1,31 @@
+   example, `"v2"`).
+
+   ```screen
+   [bos]
+   # api_version = "v2"
````
I would second this. We should be very clear about what a commented-out line looks like and what happens in that case (SAT uses its internal default). We should be clear that the line must not be commented out (i.e. not begin with `#`) in order for it to take effect.

haasken-hpe commented on docs/usage/change_bos_version.md:
````diff
@@ -0,0 +1,31 @@
+   ```
+
+If the system is using an existing SAT config file from an older version of
+SAT, the `bos` section might not exist. Instead, you should add the `bos`
````
"Instead" sounds off to me following the first sentence because the first sentence is not describing an action the user should take. I like "in that case" better.

```suggestion
SAT, the `bos` section might not exist. In that case, you should add the `bos`
```

haasken-hpe reviewed (approved):
Approving, but I will be happy to take another look after comments are addressed.

jld-dehn-hpe commented:
@haasken-hpe, @ekamin-hpe, or @jack-stanek-hpe, would one of you be willing to review the PR again before I merge? (I incorporated your changes into `change_bos_version.md` and added "uncomment" to our `.spelling` file.) I changed the flow of the instructions a fair bit and want to make sure they're accurate and understandable from a technical perspective. Thanks!

jack-stanek-hpe reviewed (approved):
Looks good to me!
